lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -336,15 +338,30 @@ ParseResult FIRParser::parseToken(FIRToken::Kind expectedToken,
   return emitError(message);
 }
 
-/// Parse a list of elements, terminated with an arbitrary token.
+/// Parse a comma-separated list of elements, terminated with an arbitrary
+/// token.
 ParseResult
 FIRParser::parseListUntil(FIRToken::Kind rightToken,
                           const std::function<ParseResult()> &parseElement) {
+  bool first = false;
```

Comment:
It might be clearer to replace this boolean state with 1-2 lines that basically unroll that path?

Reply:
I'll probably just leave this as is unless there's a need for a change. This function does what it says on the tin, and the current spelling isn't so bad.